game/atmosphere.rpy

```diff
@@ -9,87 +9,181 @@ image dim light = "mod_assets/backgrounds/classroom/dim_light.png"
+    )
+
+    WEATHER_GLITCH = JNWeather(
+        weather_type=JNWeatherTypes.glitch,
+        day_image="glitch_fuzzy")
+
+    __RANDOM_WEATHER_TYPES = [
+        WEATHER_OVERCAST,
+        WEATHER_RAIN,
+        WEATHER_THUNDER,
+        WEATHER_SUNNY
+    ]
 
     current_weather = None
 
-    def show_random_sky():
+    def show_current_sky(with_transition=True):
         """
-        Shows a randomised sky placeholder with associated dimming effect
+        Shows the sky based on the sunrise/sunset times specified under the persistent.
+
+        IN:
+            with_transition - If True, will visually fade in the new weather
         """
+        if (store.jn_get_current_hour() > store.persistent.jn_sunrise_hour 
+            and store.jn_get_current_hour() <= store.persistent.jn_sunset_hour ):
+            if store.persistent.jn_random_weather:
+                show_random_sky(with_transition=with_transition)
 
-        # Select the sky and dimming effect
-        weather_map = random.choice(list(__WEATHER_EFFECT_MAP.items()))
-        weather_type = weather_map[0]
-        sky, dim = weather_map[1]
+            elif not is_current_weather_sunny():
+                show_sky(WEATHER_SUNNY, with_transition=with_transition)
+
+    def show_random_sky(with_transition=True):
+        """
+        Shows a random sky with associated dimming effect
+
+        IN:
+            with_transition - If True, will visually fade in the new weather
+        """
+        # Select the sky
+        weather = random.choice(__RANDOM_WEATHER_TYPES)
 
         # Show the sky
-        renpy.show(name=sky, zorder=_SKY_Z_INDEX)
-        renpy.with_statement(trans=store.weather_change_transition)
+        renpy.show(name=weather.day_image, zorder=_SKY_Z_INDEX)
+        if with_transition:
+            renpy.with_statement(trans=store.weather_change_transition)
 
-        # Add the dimming effect matching the sky, if it exists
-        if dim:
-            renpy.show(name=dim, zorder=_DIM_Z_INDEX)
-            renpy.with_statement(trans=store.dim_change_transition)
+        # Add the dimming effect, if defined
+        if weather.dim_image:
+            renpy.show(name=weather.dim_image, zorder=_DIM_Z_INDEX)
+            if with_transition:
+                renpy.with_statement(trans=store.dim_change_transition)
 
         # Play rain sfx if the chosen weather is rainy
-        if weather_type in (JNWeatherTypes.rain, JNWeatherTypes.thunder):
-            renpy.music.play(filenames=__MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
+        if weather.weather_sfx:
+            renpy.music.play(filenames=weather.weather_sfx, channel="weather_loop", fadein=3.0)
+
+        else:
+            renpy.music.stop(channel="weather_loop", fadeout=5.0)
 
         global current_weather
-        current_weather = weather_type
+        current_weather = weather.weather_type
 
-    def show_sky(weather_type):
+    def show_sky(weather, with_transition=True):
```

Comment:
Might be better to make this a method of `JNWeather` to help keep things more organised.

Reply:
Could do, but unless I'm remembering wrong but we'd still have to go through the namespace, but adding a new layer of the weather class which drags things out a little more.

I personally think that weather is due an overhaul anyway to account for things like night weather (now we have a night space room with transparent windows, etc), so would rather handle it then instead of under the introduction sequence (I only made these adjustments because I needed to stop glitch being classified as a random weather!).